### PR TITLE
Improvements to Anon Names + Refactoring + Wizard Academy theme and event

### PR DESCRIPTION
**Pull request by tralezab**
#### no pref messing
Anon names now uses fully randomizing human proc instead of generating a random preference, this should preserve your character in instances where you join an anon mode game, then check your game panel and save the changes before next round

#### no longer roundstart
Anon names can now be activated at any time, it has a few more prompts for changing the crew or just making it for new joins + announcing or not

#### now can be disabled
Anon names can be DISABLED at any time, loading everyone's preferences for who they originally were. This can probably go awry, but it really doesn't need to be perfect

#### Datumized
which means it's much easier to add more themes in the future and it's way less annoying

#### Better documentation
just saying.

### New Theme:

There's one new theme for anonymous names, giving everyone a wizard name and the AI into crystallized intelligences. I'm considering refactoring this whole thing into a "ERT datum" like system so it uses less switch cases and generally lazy code

### New Event: Mass Identity Spoof
It enables anonymous names for wizards. This is pretty choice, requested by @actioninja 

## Why It's Good For The Game

I Ran Anon Names And Didn't Get An Admin Complaint. No really though there wasn't much reason to make this pregame only and it can be used for events a bit better

## Changelog
:cl: Tralezab, Nabski and Coaix helping in name string additions
add: Added the Wizard Academy anonymous theme
add: Wizard event requested by actionninja to enable the wizard anonymous theme as a summon events... event.
refactor: Themes are no longer switch cases and strings, they are now datum singletons. I really should have done this before
add: Anon names can be ran post roundstart with options to announce or not
fix: Anon names won't fuck with your preferences anymore, only loading from it instead of saving to it.
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
eff it, i'm the only one working on anon names in any capacity i'm going to refactor it to use datums and document it

File: strings/names/wizardfirst.txt [hunk 1]
```diff
@@ -1,37 +1,97 @@
...
+Tim
+Tom
 Tzeentch
+Ulysses
+Uriel
 Urza
 Vaarsuvius
 Vecna
+Vlad
+Wizardman
```
Comment (Rohesie) [inline strings/names/wizardfirst.txt:93]: what

File: strings/names/wizardsecond.txt [hunk 2]
```diff
@@ -31,10 +41,59 @@ the Sorcelator
...
+Mountain-Mother
+the Moldywarp
+Evilton
+Among-many
+Deathbringer
+Lifereaper
+Shorekeeper
+Namekeeper
+Truenamer
+Evergreen
```
Comment (Rohesie) [inline strings/names/wizardsecond.txt:80]: Could you order this alphabetically? There's phyton sort scripts and tools to automatically do that for you.

File: code/modules/admin/verbs/anonymousnames.dm [hunk 3]
```diff
@@ -1,65 +1,123 @@
...
+		priority_announce("Names and Identities have been restored.", "Identity Restoration", 'sound/ai/attention.ogg')
+		for(var/mob/living/player in GLOB.player_list)
+			if(player.mind && (ishuman(player) || issilicon(player)) && SSjob.GetJob(player.mind.assigned_role))
+				var/old_name = player.real_name //before restoration
+				if(issilicon(player))
+					var/is_AI = isAI(player)
+					player.apply_pref_name("[is_AI ? "ai" : "cyborg"]", player.client)
+				else
+					player.client.prefs.copy_to(player, antagonist = (LAZYLEN(player.mind.antag_datums) > 0), is_latejoiner = FALSE)
+					player.fully_replace_character_name(old_name, player.real_name) //this changes IDs and PDAs and whatnot
```
Comment (Rohesie) [inline code/modules/admin/verbs/anonymousnames.dm:28]:
```suggestion
			if(!player.mind || (!ishuman(player) && !issilicon(player)) || !SSjob.GetJob(player.mind.assigned_role))
				continue
			var/old_name = player.real_name //before restoration
			if(issilicon(player))
				var/is_AI = isAI(player)
				player.apply_pref_name("[is_AI ? "ai" : "cyborg"]", player.client)
			else
				player.client.prefs.copy_to(player, antagonist = (LAZYLEN(player.mind.antag_datums) > 0), is_latejoiner = FALSE)
				player.fully_replace_character_name(old_name, player.real_name) //this changes IDs and PDAs and whatnot
```
Early continue

File: code/modules/admin/verbs/anonymousnames.dm [hunk 4]
```diff
@@ -1,65 +1,123 @@
...
+ */
+/proc/anonymous_all_players()
+	var/datum/anonymous_theme/theme = SSticker.anonymousnames
+	for(var/mob/living/player in GLOB.player_list)
+		if(player.mind && (ishuman(player) || issilicon(player)) && SSjob.GetJob(player.mind.assigned_role))
+			if(issilicon(player))
+				player.fully_replace_character_name(player.real_name, theme.anonymous_ai_name(isAI(player)))
+			else
+				randomize_human(player) //do this first so the special name can be given
+				player.fully_replace_character_name(player.real_name, theme.anonymous_name(player))
```
Comment (Rohesie) [inline code/modules/admin/verbs/anonymousnames.dm:62]:
```suggestion
		if(!player.mind || (!ishuman(player) && !issilicon(player)) || !SSjob.GetJob(player.mind.assigned_role))
			continue
		if(issilicon(player))
			player.fully_replace_character_name(player.real_name, theme.anonymous_ai_name(isAI(player)))
		else
			randomize_human(player) //do this first so the special name can be given
			player.fully_replace_character_name(player.real_name, theme.anonymous_name(player))
```
Early continue

**Review by Rohesie — CHANGES_REQUESTED** [review]

**Pull request merged**